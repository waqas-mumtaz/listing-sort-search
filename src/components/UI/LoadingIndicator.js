import React from 'react';

import './LoadingIndicator.scss';

const LoadingIndicator = () => (
  <section className="section">
    <div className="container">
      <div className="columns is-centered">
        <div className="column is-half">
          <figure style={{textAlign: 'center'}}>
            <div className="lds-ring">
              <div />
              <div />
              <div />
              <div />
            </div>
          </figure>
        </div>
      </div>
    </div>
  </section>
);

export default LoadingIndicator;
