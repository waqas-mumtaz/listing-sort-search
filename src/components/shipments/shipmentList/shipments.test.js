import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ShipmentList from './shipments';
import ShipmentListItems from './shipmentListItems';
import Pagination from './pagination';
import LoadingIndicator from '../../UI/LoadingIndicator';

configure({adapter: new Adapter()});

describe('<ShipmentList />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ShipmentList />);
    });

    it('should be defined', () => {
        expect(wrapper).toBeDefined();
      });

      it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });

    it('should render <ShipmentListItems /> if data is fetched', () => {
        wrapper.setProps({dataFetched: true});
        expect(wrapper.find(ShipmentListItems)).toHaveLength(1);
    });

    it('should not render <ShipmentListItems /> if data is not fetched', () => {
        wrapper.setProps({dataFetched: false});
        expect(wrapper.find(ShipmentListItems)).toHaveLength(0);
    });

    it('should render <Pagination /> 2 times if data is fetched', () => {
        wrapper.setProps({dataFetched: true});
        expect(wrapper.find(Pagination)).toHaveLength(2);
    });

    it('should render <LoadingIndicator /> if data is not fetched', () => {
        wrapper.setProps({dataFetched: false});
        expect(wrapper.find(LoadingIndicator)).toHaveLength(1);
    });
});