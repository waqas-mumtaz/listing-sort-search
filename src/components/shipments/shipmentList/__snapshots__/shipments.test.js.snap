// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShipmentList /> should render <Pagination /> 1`] = `ShallowWrapper {}`;

exports[`<ShipmentList /> should render <Pagination /> 2 Times 1`] = `ShallowWrapper {}`;

exports[`<ShipmentList /> should render <ShipmentListItems />  1`] = `ShallowWrapper {}`;

exports[`<ShipmentList /> should render correctly 1`] = `ShallowWrapper {}`;
